mesheryctl/internal/cli/root/app/import.go

```diff
@@ -0,0 +1,281 @@
+	Short: "Import app manifests",
+	Long:  `Import the app manifest into Meshery`,
+	Args:  cobra.MinimumNArgs(0),
+	Example: `
+// Import app manifest 
+mesheryctl app import -f [file/URL] -s [source-type]
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		validTypesURL := mctlCfg.GetBaseMesheryURL() + "/api/application/types"
+		client := &http.Client{}
+		req, err := utils.NewRequest("GET", validTypesURL, nil)
+		if err != nil {
+			return err
+		}
+
+		resp, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+
+		if resp.StatusCode != 200 {
+			return errors.Errorf("Response Status Code %d, possible Server Error", resp.StatusCode)
+		}
+		defer resp.Body.Close()
+
+		var response []*models.ApplicationSourceTypesAPIResponse
+
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return errors.Wrap(err, utils.PerfError("failed to read response body"))
+		}
+		err = json.Unmarshal(body, &response)
+		if err != nil {
+			return errors.Wrap(err, "failed to unmarshal response body")
+		}
+
+		for _, apiResponse := range response {
+			validSourceTypes = append(validSourceTypes, apiResponse.ApplicationType)
+		}
+
+		return nil
+	},
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var req *http.Request
+		var err error
+		client := &http.Client{}
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return errors.Wrap(err, "error processing config")
+		}
+
+		appURL := mctlCfg.GetBaseMesheryURL() + "/api/application"
+		patternURL := mctlCfg.GetBaseMesheryURL() + "/api/pattern"
+
+		// If app file is passed via flags
+		if !isValidSource(sourceType) {
+			return errors.Errorf("application source type (-s) invalid or not passed.\nAllowed source types: %s", strings.Join(validSourceTypes, ", "))
+		}
+
+		// Check if the app manifest is file or URL
+		if validURL := govalidator.IsURL(file); !validURL {
+			content, err := os.ReadFile(file)
+			if err != nil {
+				return err
+			}
+			text := string(content)
+
+			jsonValues, err := json.Marshal(map[string]interface{}{
+				"application_data": map[string]interface{}{
+					"name":             path.Base(file),
+					"application_file": text,
+				},
+				"save": true,
+			})
+			if err != nil {
+				return err
+			}
+			req, err = utils.NewRequest("POST", appURL+"/"+sourceType, bytes.NewBuffer(jsonValues))
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Do(req)
+			if err != nil {
+				return err
+			}
+			utils.Log.Debug("app file saved")
+			var response []*models.MesheryApplication
+			// failsafe (bad api call)
+			if resp.StatusCode != 200 {
+				return errors.Errorf("Response Status Code %d, possible Server Error", resp.StatusCode)
+			}
+			defer resp.Body.Close()
+
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				return errors.Wrap(err, utils.PerfError("failed to read response body"))
```

Comment:
Is use of `PerfError` intentional?

Reply:
Uh oh 💆‍♂️ I didn't see that. Will change it

Reply:
Ok got it... I need to add a function in Meshkit for that since `appError` one is not available, doing it now